client/API.md

```diff
@@ -0,0 +1,536 @@
+| skip-format     | false   | Do not format the TICKscript before returning.                                                                                                                     |
+
+
+#### Example
+
+Get information about a task using defaults.
+
+```
+GET /task/TASK_NAME
+```
+
+```
+{
+    "name" : "TASK_NAME",
+    "type" : "stream",
+    "dbrps": [{"db": "DATABASE_NAME", "rp" : "RP_NAME"}],
+    "script": "stream\n    |from()\n        .measurement('cpu')\n",
+    "dot" : "digraph TASK_NAME { ... }",
+    "enabled" : true,
+    "executing" : true,
+    "error" : "",
+    "stats": {}
+}
+```
+
+Get information about a task using only labels in the DOT content and skip the format step.
+
+```
+GET /task/TASK_NAME?skip-format=true&dot-labels=true
+```
+
+```
+{
+    "name" : "TASK_NAME",
+    "type" : "stream",
+    "dbrps": [{"db": "DATABASE_NAME", "rp" : "RP_NAME"}],
+    "script": "stream|from().measurement('cpu')",
+    "dot" : "digraph TASK_NAME { ... }",
+    "enabled" : true,
+    "executing" : true,
+    "error" : "",
+    "stats": {}
+}
+```
+
+#### Response
+
+| Code | Meaning             |
+| ---- | -------             |
+| 200  | Success             |
+| 404  | Task does not exist |
+
+
+### Enable Task
+
+To enable a task and start it executing make a POST request to the `/task/TASK_NAME/enable` endpoint.
+
+
+#### Example
+
+```
+POST /task/TASK_NAME/enable
+```
+
+#### Response
+
+| Code | Meaning             |
+| ---- | -------             |
+| 204  | Success             |
+| 404  | Task does not exist |
+
+### Disable Task
+
+To disable a task and stop it executing make a POST request to the `/task/TASK_NAME/disable` endpoint.
+
+#### Example
+
+```
+POST /task/TASK_NAME/disable
+```
+
+#### Response
+
+| Code | Meaning             |
+| ---- | -------             |
+| 204  | Success             |
+| 404  | Task does not exist |
+
+
+### Delete Task
+
+To delete a task make a DELETE request to the `/task/TASK_NAME` endpoint.
+
+```
+DELETE /task/TASK_NAME
+```
+
+#### Response
+
+| Code | Meaning             |
+| ---- | -------             |
+| 204  | Success             |
+
+>NOTE: Deleting a non-existent task is not an error and will return a 204 success.
+
+
+### List Tasks
+
+To get information about several tasks make a GET request to the `/task` endpoint.
+
+| Query Parameter | Purpose                                                                                                                                           |
+| --------------- | -------                                                                                                                                           |
+| pattern         | Filter results based on the pattern. Uses standard shell glob matching, see [this](https://golang.org/pkg/path/filepath/#Match) for more details. |
+
+#### Example
+
+Get all tasks.
+
+```
+GET /task
```

Comment:
This should have pagination when there are many tasks.  So,  I think it should have RFC5988 links:

```http
GET /task HTTP/1.1

Link: <https://kapa.com/task?offset=20&limit=10>; rel="next",<https://kapa.com/task?offset=10&limit=10>; rel="self",<https://kapa.com/task?offset=0&limit=10>; rel="prev",
```

In addition to RFC5988 header links, some REST interfaces also put a `link` object in the response:

```json
"links" : [
  {"rel": "next", "href": "https://kapa.com/task?offset=20&limit=10"},
  {"rel": "prev", "href": "https://kapa.com/task?offset=0&limit=10"},
  {"rel": "self", "href": "https://kapa.com/task?offset=10&limit=10"},
]
```
... I find these extra link sections a bit too verbose, but, it can be nice for moving through pages in browser.